subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
That sounds reasonable to me. 👍 

To be clear on the ordering of things that need to happen: Does that mean we'll have to deprecate 1.5 first?